networking/networking.go

```diff
@@ -50,6 +54,114 @@ type Networking struct {
 	nets   []activeNet
 }
 
+type NetConf struct {
+	plugin.NetConf
+	IPMasq bool `json:"ipMasq"`
+	MTU    int  `json:"mtu"`
+}
+
+func setupTapDevice() (string, *netlink.Link, error) {
+	ifName, err := tuntap.CreatePersistentIface(tuntap.Tap)
+	if err != nil {
+		return "", nil, fmt.Errorf("tuntap persist %v", err)
+	}
+	var link netlink.Link
+	link, err = netlink.LinkByName(ifName)
+	if err != nil {
+		return "", nil, fmt.Errorf("cannot find link %q: %v", ifName, err)
+	}
+	err = netlink.LinkSetUp(link)
+	if err != nil {
+		return "", nil, fmt.Errorf("cannot set link up %q: %v", ifName, err)
+	}
+	return ifName, &link, nil
+}
+
+func kvmSetupNetAddressing(network *Networking, n activeNet, ifName string) error {
+	// TODO: very ugly hack, that go through upper plugin, down to ipam plugin
```

Comment:
TODO: needs better explanation - "very ugly hack" doesn't helps a lot - I forgot what it means even I wrote this two weeks ago